mixer/adapter/stackdriver/contextgraph/send.go

```diff
@@ -90,6 +90,27 @@ func (e edge) ToProto() *contextgraphpb.Relationship {
 	}
 }
 
+// splitBySize helps find the largest BatchRequest that is smaller than the max request size
+func splitBySizeEntity(msg *contextgraphpb.AssertBatchRequest) int {
+	curSize := proto.Size(msg)
```

Comment:
This still seems to count size of each assertion for multiple times. How about this:
```
curIndex := 0
curSize := 0
for ; curIndex < len(msg.EntityPresentAssertions); curIndex++ {
  curSize += proto.Size(msg.EntityPresentAssertions[curIndex])
  if curSize >= maxReq {
    break
  }
}
return curIndex
```

Reply:
Yeah that should be a little quicker, updated.